discovery-handlers/udev/src/discovery_impl.rs

```diff
@@ -543,6 +543,24 @@ fn device_or_parents_have_tag(device: &impl DeviceExt, value_regex: &Regex) -> b
     }
 }
 
+/// Retrieve Parent or Children of a device using their sysfs path.
+pub fn get_device_relatives<'a>(
```

Comment:
@diconico07 can you write a few unit tests for this?

Reply:
I wrote a test case, tell me if you feel I missed something :smile: